src/dbnode/storage/index/aggregate_results.go

```diff
@@ -123,6 +123,7 @@ func (r *aggregatedResults) AggregateResultsOptions() AggregateResultsOptions {
 
 func (r *aggregatedResults) AddFields(batch []AggregateResultsEntry) (int, int) {
 	r.Lock()
+	maxInsertions := r.aggregateOpts.SizeLimit - r.totalDocsCount
```

Comment:
should we be subtracting totalDocsCount or r.resultsMap.size()?

Reply:
Yeah going to do some refactoring/renaming around this to make it clearer what each limit is; totalDocsCount is not _quite_ correctly calculated at the moment, so will need to make a few touchups around it